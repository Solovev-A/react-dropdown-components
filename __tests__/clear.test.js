import React from 'react';
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import 'jest-styled-components'

import Clear from '../lib/components/clear';

describe('Clear component', () => {
    it('renders correctly', () => {
        const { container } = render(<Clear />);
        expect(container.firstChild).toMatchSnapshot();
    });

    describe('behavior', () => {
        it('use callback on click', () => {
            const onClick = jest.fn();
            const { container } = render(<Clear onClick={onClick} />);
            userEvent.click(container.firstChild);
            expect(onClick).toHaveBeenCalledTimes(1);
        });
    });
});