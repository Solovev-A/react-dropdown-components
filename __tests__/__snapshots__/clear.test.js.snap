// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clear component renders correctly 1`] = `
.c0 {
  padding-right: 3px;
  padding-left: 3px;
  font-weight: 700;
  font-size: 1.25rem;
  color: #bdc6d0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0:hover {
  color: #343a40;
}

<span
  class="c0"
  title="Очистить"
>
  ×
</span>
`;
